x/pricefeed/abci.go

```diff
@@ -10,19 +10,14 @@ import (
 func EndBlocker(ctx sdk.Context, k Keeper) {
 	// Update the current price of each asset.
 	for _, market := range k.GetMarkets(ctx) {
-		if market.Active {
-			err := k.SetCurrentPrices(ctx, market.MarketID)
-			if err != nil {
-				// In the event of failure, emit an event.
-				ctx.EventManager().EmitEvent(
-					sdk.NewEvent(
-						EventTypeNoValidPrices,
-						sdk.NewAttribute(AttributeMarketID, fmt.Sprintf("%s", market.MarketID)),
-					),
-				)
-				continue
-			}
+		if !market.Active {
+			continue
+		}
+
+		err := k.SetCurrentPrices(ctx, market.MarketID)
+		if err != nil {
+			// TODO: this should panic
```

Comment:
If we want to panic here, we should check that the error is not `ErrNoValidPrice`. All oracles going down is bad, but shouldn't cause an application panic.